### PR TITLE
Core: Check for Dread Spikes/Reprisal effect before getting subpower

### PR DESCRIPTION
**Pull request by Hozu**
Just in case... as recommended by @teschnei

### CODE REVIEW

**Comment by teschnei** [comment]
:+1:

File: src/map/utils/battleutils.cpp [hunk 1]
```diff
@@ -730,15 +730,19 @@ namespace battleutils
...
+                            auto PEffect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES);
+                            if (PEffect)
                             {
-                                PDefender->StatusEffectContainer->DelStatusEffect(EFFECT_DREAD_SPIKES);
-                            }
-                            else
-                            {
-                                PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->SetSubPower(remainingDrain - Action->spikesParam);
+                                // Subpower is the remaining damage that can be drained. When it reaches 0 the effect ends
+                                int remainingDrain = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_DREAD_SPIKES)->GetSubPower();
```
Comment (bendangelo) [inline src/map/utils/battleutils.cpp]: Can you use `PEffect` here and in other places?

File: src/map/utils/battleutils.cpp [hunk 2]
```diff
@@ -750,16 +754,19 @@ namespace battleutils
...
-                            PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL)->SetSubPower(remainingReflect - Action->spikesParam);
+                            // Subpower is the remaining damage that can be reflected. When it reaches 0 the effect ends
+                            int remainingReflect = PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL)->GetSubPower();
+                            if (remainingReflect - Action->spikesParam <= 0)
+                            {
+                                PDefender->StatusEffectContainer->DelStatusEffect(EFFECT_REPRISAL);
+                            }
+                            else
+                            {
+                                PDefender->StatusEffectContainer->GetStatusEffect(EFFECT_REPRISAL)->SetSubPower(remainingReflect - Action->spikesParam);
```
Comment (bendangelo) [inline src/map/utils/battleutils.cpp]: `PEffect->SetSubPower` Thank you.

File: src/map/utils/battleutils.cpp (hunk 1 above)
Comment (Hozu) [inline src/map/utils/battleutils.cpp]: You mean PEffect->>GetSubPower(); ?

Comment (bendangelo) [inline src/map/utils/battleutils.cpp]: Yep

**Comment by Hozu** [comment]
Ok made the changes requested by @bendangelo and tested, works properly still.

**Comment by bendangelo** [comment]
thanks

**Pull request merged**